modules/google-analytics/wp-google-analytics.php

```diff
@@ -0,0 +1,408 @@
+			$custom_vars[] = "_gaq.push(['_trackPageview','{$track['url']}']);";
+		} else {
+			$custom_vars[] = "_gaq.push(['_trackPageview']);";
+		}
+
+		if ( 'true' === $wga['enable_display_advertising'] ) {
+			$async_code = "<script type='text/javascript'>
+								var _gaq = _gaq || [];
+								%custom_vars%
+
+								(function() {
+									var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
+									ga.src = ('https:' == document.location.protocol ? 'https://' : 'http://') + 'stats.g.doubleclick.net/dc.js';
+									var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
+								})();
+							</script>";
+		} else {
+			$async_code = "<script type='text/javascript'>
+								var _gaq = _gaq || [];
+								%custom_vars%
+
+								(function() {
+									var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
+									ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
+									var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
+								})();
+							</script>";
+		}
+
+		$custom_vars_string = implode( "\r\n", $custom_vars );
+		$async_code = str_replace( '%custom_vars%', $custom_vars_string, $async_code );
+
+		return $this->_output_or_return( $async_code, $output );
+
+	}
+
+	/**
+	 * Used to get one or all of our plugin options
+	 *
+	 * @param string[optional] $option - Name of options you want.  Do not use if you want ALL options.
+	 * @param boolean[optiona] $default - Default value.
+	 * @return array of options, or option value.
+	 */
+	private function _get_options( $option = null, $default = false ) {
+
+		$o = get_option( 'jetpack_wga' );
+
+		if ( isset( $option ) ) {
+			if ( isset( $o[ $option ] ) ) {
+				if ( 'code' === $option ) {
+					if ( preg_match( '#UA-[\d-]+#', $o[ $option ], $matches ) ) {
+						return $matches[0];
+					} else {
+						return '';
+					}
+				} else {
+					return $o[ $option ];
+				}
+			} else {
+				if ( 'ignore_role_' === substr( $option, 0, 12 ) ) {
+					global $wp_roles;
+					// Backwards compat for when the tracking information was stored as a cap.
+					$maybe_role = str_replace( 'ignore_role_', '', $option );
+					if ( isset( $wp_roles->roles[ $maybe_role ] ) ) {
+						if ( isset( $wp_roles->roles[ $maybe_role ]['capabilities']['wga_no_track'] ) && $wp_roles->roles[ $maybe_role ]['capabilities']['wga_no_track'] ) {
+							return 'true';
+						}
+					}
+					return false;
+				}
+				return $default;
+			}
+		} else {
+			return $o;
+		}
+	}
+
+	/**
+	 * If we track outgoing links, this will enqueue our javascript file
+	 */
+	public function track_outgoing() {
+		if ( 'true' === $this->_get_options( 'log_outgoing' ) && ( ! defined( 'XMLRPC_REQUEST' ) || ! XMLRPC_REQUEST ) && ( ! is_admin() || 'false' === $this->_get_options( 'ignore_admin_area' ) ) ) {
```

Comment:
Will this option bet set anywhere in Calypso? If not, we could probably remove all this, along with the associated js file.